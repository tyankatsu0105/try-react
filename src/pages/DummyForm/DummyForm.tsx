import React, { FunctionComponent } from 'react';
import { Helmet } from 'react-helmet';
import {
  Box,
  Form as GForm,
  FormField,
  TextInput,
  TextArea,
  Button,
} from 'grommet';

const DummyForm: FunctionComponent = () => {
  return (
    <>
      <Helmet>
        <title>This is DummyForm</title>
      </Helmet>

      <GForm>
        <FormField
          label="名前"
          name="name"
          placeholder="田中　太郎"
          component={TextInput}
          validate={{ regexp: /hoge/, message: 'hogeのみ受け付けます' }}
        />
        <FormField
          label="詳細"
          name="description"
          placeholder="長いテキストが入る"
          component={TextArea}
          validate={{ regexp: /fuga/, message: 'fugaのみ受け付けます' }}
        />
        <Box align="center">
          <Button type="submit" primary label="Submit" />
        </Box>
      </GForm>
    </>
  );
};

export default DummyForm;
