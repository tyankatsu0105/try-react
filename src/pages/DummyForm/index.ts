import { lazy } from 'react';

export const DummyForm = lazy(() => import('./DummyForm'));
