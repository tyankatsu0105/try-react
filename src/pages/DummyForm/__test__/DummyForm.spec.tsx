import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Grommet } from 'grommet';

import DummyForm from '../DummyForm';

describe('DummyForm', () => {
  it('snapshot', () => {
    const { baseElement } = render(
      <MemoryRouter initialEntries={['/']}>
        <Grommet>
          <DummyForm />
        </Grommet>
      </MemoryRouter>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
