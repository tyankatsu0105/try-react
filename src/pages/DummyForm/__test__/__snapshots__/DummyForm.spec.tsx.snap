// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DummyForm snapshot 1`] = `
<body>
  <div>
    <div
      class="StyledGrommet-sc-19lkkz7-0 eUfLDp"
    >
      <form>
        <div
          class="FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
        >
          <label
            class="StyledText-sc-1sadyjn-0 bKLvLm"
          >
            名前
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gZSidW"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kYQCpm"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 duvNfq"
                label="名前"
                name="name"
                placeholder="田中　太郎"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="FormField__FormFieldBox-m9hood-0 eshLKp StyledBox-sc-13pk1d4-0 CAsMy"
        >
          <label
            class="StyledText-sc-1sadyjn-0 bKLvLm"
          >
            詳細
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 gZSidW"
          >
            <textarea
              class="StyledTextArea-sc-17i3mwp-0 gIdccB"
              label="詳細"
              name="description"
              placeholder="長いテキストが入る"
            />
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 dNgQiZ"
        >
          <button
            class="StyledButton-sc-323bzc-0 aNlbA"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</body>
`;
